src/tensors/braidingtensor.jl

```diff
@@ -209,30 +210,40 @@ end
 # TensorOperations
 # ----------------
 # TODO: implement specialized methods
-function TO.tensoradd!(C::AbstractTensorMap{S,N₁,N₂}, pC::Index2Tuple{N₁,N₂},
-                       A::BraidingTensor{S}, conjA::Symbol, α::Number, β::Number,
-                       backend::Backend...) where {S,N₁,N₂}
-    return TO.tensoradd!(C, pC, copy(A), conjA, α, β, backend...)
+
+function TO.tensoradd!(C::AbstractTensorMap,
+                       A::BraidingTensor, pA::Index2Tuple, conjA::Symbol,
+                       α::Number, β::Number, backend::AbstractBackend=TO.DefaultBackend(),
+                       allocator=TO.DefaultAllocator())
+    return TO.tensoradd!(C, TensorMap(A), pA, conjA, α, β, backend, allocator)
 end
 
 # Planar operations
 # -----------------
-function planaradd!(C::AbstractTensorMap{S,N₁,N₂},
-                    A::BraidingTensor{S},
-                    p::Index2Tuple{N₁,N₂},
+# TODO: implement specialized methods
+
+function planaradd!(C::AbstractTensorMap,
+                    A::BraidingTensor, p::Index2Tuple,
                     α::Number, β::Number,
-                    backend::Backend...) where {S,N₁,N₂}
-    return planaradd!(C, copy(A), p, α, β, backend...)
+                    backend::AbstractBackend,
+                    allocator)
+    return planaradd!(C, TensorMap(A), p, α, β, backend, allocator)
 end
 
-function planarcontract!(C::AbstractTensorMap{S,N₁,N₂},
-                         A::BraidingTensor{S},
-                         (oindA, cindA)::Index2Tuple{2,2},
-                         B::AbstractTensorMap{S},
-                         (cindB, oindB)::Index2Tuple{2,N₃},
-                         (p1, p2)::Index2Tuple{N₁,N₂},
+function planarcontract!(C::AbstractTensorMap,
+                         A::BraidingTensor,
+                         (oindA, cindA)::Index2Tuple,
```

Comment:
I am confused how this can be valid without the specific `(oindA, cindA)::Index2Tuple{2,2}` part? The implementation for `Bosonic` below really assumes that you are exactly permuting two indices, and not have e.g. 3 indices contracted with the braiding tensor (corresponding to a trace operation).

Reply:
Oh I see, you've added some check below. Never mind